src/components/company-data-table.tsx

```
@@ -16,433 +16,519 @@ import { CompanyNameBadge } from './company-name-badge';
-                return formatNumber ? `${formatNumber(numValue, { decimals: 1 })} yrs` : String(numValue) + ' yrs';
-            case 'ratio':
-                return formatNumber ? formatNumber(numValue, { decimals: 2 }) : String(numValue);
-            default:
-                return String(value);
+  if (value === null || value === undefined) return '-';
+  const numValue = Number(value);
+  
+  // Special handling for formats that don't require isFinite check (like 'string')
+  if (format === 'string') return String(value);
+
+  if (!isFinite(numValue)) return '-'; // Apply for numeric formats after string check
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)